mlflow/pyfunc/__init__.py

```diff
@@ -1484,7 +1506,10 @@ def batch_predict_fn(pdf, params=None):
                 return client.invoke(pdf).get_predictions()
 
         elif env_manager == _EnvManager.LOCAL:
-            if should_use_spark_to_broadcast_file:
+            if is_spark_connect_session:
+                with tempfile.TemporaryDirectory() as tmpdir:
+                    loaded_model = mlflow.pyfunc.load_model(model_uri, dst_path=tmpdir)
```

Comment:
We can do a check:

if running code on databricks notebook, then mlflow still can use NFS directory in spark connect mode.

Reply:
For shared cluster, the databricks python REPL will also run as spark connect mode

Reply:
sounds good